src/pieces/piece.ts

```diff
@@ -1,28 +1,33 @@
 import board from '../board';
 
-class Piece {
-    constructor(public x: number, public y: number, private side: string) {
+abstract class Piece {
+    display: string;
+
+    constructor(public x: number, public y: number, protected side: string) {
         this.x = x;
         this.y = y;
-        this.side = side; //'black' or 'white'
+        this.side! = side; //'black' or 'white'
+        this.display = '';
     }
     //id to tablica z kliknieta pozycja na ktra pionek ma sie poruszyc  
-    move(id: Array<string>) {
+    move(id: string) {
         const newX = Number(id[0]);
         const newY = Number(id[2]);
 
         //clearing previous place
         board[this.x][this.y] = null;
-        document.getElementById(`${this.x},${this.y}`).innerHTML = '';
+        (document.getElementById(`${this.x},${this.y}`) as HTMLDivElement).innerHTML = '';
         // debugger
         //setting new
         this.x = newX;
         this.y = newY;
         board[this.x][this.y] = this;
-        document.getElementById(id).innerHTML = this.display;
+        (document.getElementById(id) as HTMLDivElement).innerHTML = this.display;
     }
 
-    // findLegalMoves() { }
+    findLegalMoves(): any { 
```

Comment:
Typo, tutaj nie powinno być raczej any

Reply:
tak nie używamy any, bo potem dzieją sie cyrki w  programie any mówi TS-owi  rób co chcesz a my nie chcemy takiej sytuacji bo to  sie dzieje w czystym JS-ie. Na ten moment możemy ją wykomentować albo tak zostawić bo funkcja nic nie robi ale jak wybierzesz drugą opcją to dodaj nad funkcję komentarz to  refactor when findLegalMoves will bo change lub jakiś  podobny żeby nam nie umkneło

Reply:
Ok poprawię, racja mogłem dodać komentarz bo to było tylko rozwiązanie tymczasowe i zapomniałem o tym ;p

Reply:
Zostawiłem na razie any z komentarzem, żeby zmienić przy implementacji metody